json/src/fs2/data/json/package.scala

```diff
@@ -51,8 +51,36 @@ package object json {
   def transform[F[_], Json](selector: Selector, f: Json => Json)(implicit F: RaiseThrowable[F],
                                                                  builder: Builder[Json],
                                                                  tokenizer: Tokenizer[Json]): Pipe[F, Token, Token] =
+    TokenSelector.transformPipe[F, Json](selector, f.andThen(Option(_)))
+
+  /** Transforms a stream of token into another one. The transformation function `f` is
+    * called on every selected value from upstream, and the resulting value replaces it.
+    * If the function returns `None`, then the entire value is dropped (and the object key it
+    * is located at, if any).
+    * The rest of the stream is left unchanged.
+    *
+    * This operator locally creates Json AST values using the [[Builder]], and
+    * returns tokens as emitted by the [[Tokenizer]] on the resulting value.
+    */
+  def transformOpt[F[_], Json](selector: Selector, f: Json => Option[Json])(implicit F: RaiseThrowable[F],
+                                                                 builder: Builder[Json],
+                                                                 tokenizer: Tokenizer[Json]): Pipe[F, Token, Token] =
     TokenSelector.transformPipe[F, Json](selector, f)
 
+  /** Transforms a stream of token into another one. The transformation function `f` is
+    * called on every selected value from upstream, and the resulting value replaces it.
+    * The rest of the stream is left unchanged. The operation can fail, in case the returned
+    * `F` is failed at one step.
+    *
+    * This operator locally creates Json AST values using the [[Builder]], and
+    * returns tokens as emitted by the [[Tokenizer]] on the resulting value.
+    */
+  def transformF[F[_], Json](selector: Selector, f: Json => F[Json])(
```

Comment:
Just wondered whether a `transformOptF` would make sense for completeness? Or thinking even more general, something like 
```scala
def flatTransform[F[_], Json](selector: Selector, f: Json => Stream[F, Json])(...)
```

Reply:
Oh, I guess the `flatTransform` could break the JSON structure in some cases…so maybe only `transformOptF`

Reply:
Yes, I'd like these pipes to emit valid sequence of tokens if the input steam was valid. Returning a stream here cannot guarantee this. I also thought about `transformOptF` and the more I think about it, the more I think I should do for completeness sake, but I wonder whether I want the `F` versions at all ^^